extensions/wikia/WallNotifications/WallNotifications.class.php

```diff
@@ -370,12 +370,11 @@ protected function createKeyForMailNotification( $watcher, WallNotificationEntit
 	}
 
 	protected function sendEmails( array $watchers, WallNotificationEntity $notification ) {
-		$text = strip_tags( $notification->data_noncached->msg_text, '<p><br>' );
+		$text = preg_replace( array( '#<br\s*/?>#i', '#</p\s*/?>#i' ), "\n",
+			$notification->data_noncached->msg_text );
+		$text = trim( strip_tags( $text ) );
 		$text = substr( $text, 0, 3000 ) . ( strlen( $text ) > 3000 ? '...' : '' );
-
-		$textNoHtml = preg_replace( '#<br\s*/?>#i', "\n", $text );
-		$textNoHtml = trim( preg_replace( '#</?p\s*/?>#i', "\n", $textNoHtml ) );
-		$textNoHtml = substr( $textNoHtml, 0, 3000) . ( strlen( $textNoHtml ) > 3000 ? '...' : '' );
+		$textNoNewlines = str_replace( "\n", ' ', $text );
```

Comment:
The original code didn't strip out the newlines; is there a reason to do that here?  Also, the variable as used still contained the ```<br>``` and ```<p>``` tags.  It seems like ```$textNoHtml``` was probably the right (intended result) name but they just didn't do a good job with the regex since it doesn't account for parameters for the ```<p>``` tag.

Maybe the we should take the original, use the array form of ```preg_replace``` you introduced and then add a strip_tags afterward.